modules/router/templates/assets_origin.conf.erb

```diff
@@ -49,7 +49,16 @@ server {
 
   <%- @asset_manager_uploaded_assets_routes.each do |path| -%>
   location <%= path %> {
-    proxy_pass <%= @upstream_ssl ? 'https' : 'http' %>://static.<%= @app_domain %>;
+    # Explicitly re-include Strict-Transport-Security header, this
+    # forces nginx not to clear Cache-Control headers further up the
+    # stack.
+    include /etc/nginx/add-sts.conf;
+
+    add_header "Access-Control-Allow-Origin" "*";
+    add_header "Access-Control-Allow-Methods" "GET, OPTIONS";
+    add_header "Access-Control-Allow-Headers" "origin, authorization";
+
+    proxy_pass <%= @upstream_ssl ? 'https' : 'http' %>://asset-manager.<%= @app_domain %>;
```

Comment:
Minor: I think it might be worth explaining in the commit note that `@upstream_ssl` incorporates the logic that was in `@enable_ssl || @proxy_pass_asset_manager_host_https`:

* https://github.com/alphagov/govuk-puppet/blob/c68837ae461461319af90df0040fb4c902ccec64/modules/router/manifests/assets_origin.pp#L38-L44
* https://github.com/alphagov/govuk-puppet/blob/c68837ae461461319af90df0040fb4c902ccec64/modules/govuk/manifests/apps/static.pp#L59-L63

Reply:
I've updated the commit note as you've suggested.